plugins/woocommerce/includes/admin/class-wc-admin-settings.php

```diff
@@ -213,6 +213,9 @@ public static function output_fields( $options ) {
 				if ( ! isset( $value['id'] ) ) {
 					$value['id'] = '';
 				}
+				if ( ! isset( $value['name'] ) ) {
+					$value['name'] = $value['id'];
+				}
 				if ( ! isset( $value['title'] ) ) {
 					$value['title'] = isset( $value['name'] ) ? $value['name'] : '';
```

Comment:
The problem I [noted here](https://github.com/woocommerce/woocommerce/pull/32728#pullrequestreview-986847278) seems to relate to this existing use of the `'name'` key.

Reply:
This block has not been modified by the commits of this pull request.
Before this PR, `"name"` key was not used, so setting `$value['name']` to `$value['title']` by default really made sense? 
Why it was done like that? Maybe you can enlighten me on this point 💡

So I think that if `$value['title']` is not defined, then it should just contain an empty string, and no default value.

Reply:
Yes, it's existing code.

> Before this PR, "name" key was not used ... Why it was done like that? Maybe you can enlighten me on this point 

It's hard to know, without doing some digging. An artifact of the past?

> So I think that if $value['title'] is not defined, then it should just contain an empty string, and no default value.

Possibly, but we also need to be mindful of compatibility with third party code. Where possible, we don't want to trigger breakages for extensions and custom code relying on existing functionality.

Reply:
> An artifact of the past?

It seems to be something like this, possibily!

> Possibly, but we also need to be mindful of compatibility with third party code. Where possible, we don't want to trigger breakages for extensions and custom code relying on existing functionality.

Yes I understand completely. Should the new "name" key we have been working on have another name? To avoid colliding with this old block? 
I find it quite unfortunate because the name is perfectly appropriate, and change it to another may be confusing for other developer in the future.

Reply:
Couldn't this code block be reworked? We could warn the third party codes through the release note in the next update if this pull request is merged. What do you think?

Reply:
That's a possibility, but I think the safer course of action is leaving it in place and switching to `'field_name'` or other suitable alternative for this new role (complete with an explanatory comment, referencing the existing usage of `'name'`).

* The existing use of `'name'`has been in place since at least 3.0.0 (and *probably* earlier).
* If we don't remove it, the two usages of `'name'` will collide with one another (as in the problem [noted here](https://github.com/woocommerce/woocommerce/pull/32728#pullrequestreview-986847278)).
* If we remove it, as in [this change](https://github.com/woocommerce/woocommerce/pull/32728/commits/93e1dd6af44a98e0217802e96f093b63b3b08eca), then we risk two different types of breakage:
    * Titles not rendering as expected in some cases (less serious).
    * Settings being saved under a different-than-expected key (more serious).

I don't think a mention in the next set of release notes is the right trade-off here.

Reply:
> Settings being saved under a different-than-expected key (more serious).

Ok sure, I just understood. If users had created fields a long time ago using the `'name'` key, indeed there could be conflict troubles with the changes we are making to that key.
So yes, _our_ `'name'` key should be renamed in `'field_name`', that seems appropriate.